include/igraph_error.h

```diff
@@ -563,7 +563,11 @@ DECLDIR int IGRAPH_FINALLY_STACK_SIZE(void);
 #define IGRAPH_FINALLY(func,ptr) \
   IGRAPH_FINALLY_REAL((igraph_finally_func_t*)(func), (ptr))
 
-#if (defined(__GNUC__) && GCC_VERSION_MAJOR >= 3)
+#if !defined(GCC_VERSION_MAJOR) && defined(__GNUC__)
+#  define GCC_VERSION_MAJOR  __GNUC__
```

Comment:
Not sure whether this is correct.  AFAIK `__GNUC__` is just a "marker" that is specified when we are using the GNU C compiler. `GCC_VERSION_MAJOR` contains the major version of the GCC compiler. Simply defining `GCC_VERSION_MAJOR` as `__GNUC__` seems fishy.

Which compiler were you using that defined `__GNUC__` but not `GCC_VERSION_MAJOR`?

Reply:
"GCC version x.y.z defines `__GNUC__` to x, `__GNUC_MINOR__` to y, and `__GNUC_PATCHLEVEL__` to z.", see the [official GCC docs](https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html).
I used `gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.12)` on Ubuntu 16.04 x64.

Reply:
Hmmm, indeed. How on Earth could this have worked before?